node/src/lib.rs

```diff
@@ -280,8 +280,104 @@ impl Node {
         Node::with_conf(exe, &Conf::default())
     }
 
-    /// Launch the bitcoind process from the given `exe` executable with given [Conf] param and create/load the "default" wallet.
+    /// Launch the bitcoind process from the given `exe` executable with given [Conf] param and
+    /// create/load the "default" wallet.
+    /// Waits for the node to be ready to accept connections before returning.
+    ///
+    /// # Parameters
+    ///
+    /// * `exe` - The path to the bitcoind executable.
+    /// * `conf` - The configuration parameters for the node.
+    ///
+    /// # Returns
+    ///
+    /// A [`Node`] instance if the node is successfully started and ready to accept connections.
+    ///
+    /// # Errors
+    ///
+    /// If the node fails to start after the specified number of attempts.
     pub fn with_conf<S: AsRef<OsStr>>(exe: S, conf: &Conf) -> anyhow::Result<Node> {
+        for attempt in 0..conf.attempts {
+            let work_dir = Self::init_work_dir(conf)?;
+            let cookie_file = work_dir.path().join(conf.network).join(".cookie");
+
+            let rpc_port = get_available_port()?;
+            let rpc_socket = SocketAddrV4::new(LOCAL_IP, rpc_port);
+            let rpc_url = format!("http://{}", rpc_socket);
+
+            let (p2p_args, p2p_socket) = Self::p2p_args(&conf.p2p)?;
+            let (zmq_args, zmq_pub_raw_tx_socket, zmq_pub_raw_block_socket) =
+                Self::zmq_args(conf.enable_zmq)?;
+
+            let stdout = if conf.view_stdout { Stdio::inherit() } else { Stdio::null() };
+
+            let datadir_arg = format!("-datadir={}", work_dir.path().display());
+            let rpc_arg = format!("-rpcport={}", rpc_port);
+            let default_args = [&datadir_arg, &rpc_arg];
+            let conf_args = validate_args(conf.args.clone())?;
+
+            let mut process = Command::new(exe.as_ref())
+                .args(default_args)
+                .args(&p2p_args)
+                .args(&conf_args)
+                .args(&zmq_args)
+                .stdout(stdout)
+                .spawn()
+                .with_context(|| format!("Error while executing {:?}", exe.as_ref()))?;
+            match process.try_wait() {
+                Ok(Some(_)) | Err(_) => {
+                    // Process has exited or an error occurred, kill and retry
+                    let _ = process.kill();
+                    continue;
+                }
+                Ok(None) => {
+                    // Process is still running, proceed
+                }
+            }
+
+            if Self::wait_for_cookie_file(cookie_file.as_path(), Duration::from_secs(5)).is_err() {
+                let _ = process.kill();
```

Comment:
Maybe a code comment here saying why we kill the process if waiting for the cookie file fails?

Reply:
I have added comments to all the checks where it kills the process and retries.